jsonargparse_tests/test_typehints.py

```diff
@@ -605,6 +609,76 @@
     ctx.match("Expected a <class 'int'>")
 
 
+@pytest.mark.skipif(not Unpack, reason="Unpack introduced in python 3.11 or backported in typing_extensions")
+def test_unpack_support(parser):
+    assert ActionTypeHint.is_supported_typehint(Unpack[Any])
+
+
+if Unpack:  # and Required and NotRequired
+    MyTestUnpackDict = TypedDict("MyTestUnpackDict", {"a": Required[int], "b": NotRequired[int]}, total=True)
+
+    class UnpackClass:
+
+        def __init__(self, **kwargs: Unpack[MyTestUnpackDict]) -> None:
+            self.a = kwargs["a"]
+            self.b = kwargs.get("b")
+
+    # force fallback on AST resolver
+    UnpackByAssumptionsClass = type(
+        "UnpackByAssumptionsClass",
+        (UnpackClass,),
+        {"__module__": __name__, "__init__": lambda self, **kwargs: UnpackClass.__init__(self, **kwargs)},
```

Comment:
The lambda makes a functional difference in the parameter resolution. Replacing it with simply `UnpackClass.__init__` would remove `jsonargparse._parameter_resolvers.get_parameters_by_assumptions` from `test_unpack_typeddict_by_assumptions`'s test coverage. 

Alternatively, using a locally defined function 
```python 
def init_by_assumptions(s, **kwargs):
    UnpackClass.__init__(s, **kwargs)
```
yields an `ArgumentError` due to mismatched expectations in the AST parameter resolver. In particular, the resolver ignores the `s` argument given to `UnpackClass.__init__` and removes the otherwise correctly inferred `a` parameter in `jsonargparse._parameter_resolvers.remove_given_parameters`.

Reply:
To create a test that uses the parameter by assumptions should be like this https://github.com/omni-us/jsonargparse/blob/42d7a5c2139583bf8cec1aa79e6afe788f878385/jsonargparse_tests/test_parameter_resolvers.py#L570

Also not good for a test to use a class created in an unusual way, since people don't do this. Could be that on real situations things might not work as expected, and the test doesn't capture it because of the unusual class.

But anyway, I think we shouldn't have support for the unpack typeddict in the parameters by assumptions.